### ISSUE TITLE
Template not found for Macro with Nunjucks environment

### ISSUE THREAD

**Issue by MKlblangenois** [11ty/eleventy#1935]
**Describe the bug**
Eleventy throw an error when I define `Nunjucks` environment with macros.

```bash
Problem writing Eleventy templates: (more in DEBUG output)
> Having trouble rendering njk template ./src/pages/index.njk

`TemplateContentRenderError` was thrown
> (./src/pages/index.njk)
  Error: template not found: macros/organisms/louds/loudfrontpage.njk

Template render error` was thrown:
    Template render error: (./src/pages/index.njk)
      Error: template not found: macros/organisms/louds/loudfrontpage.njk
        at Object._prettifyError (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/lib.js:36:11)
        at /Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/environment.js:563:19
        at eval (eval at _compile (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/environment.js:633:18), <anonymous>:12:11)
        at createTemplate (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/environment.js:295:11)
        at next (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/lib.js:330:7)
        at handle (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/environment.js:329:11)
        at /Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/environment.js:339:9
        at next (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/lib.js:328:7)
        at Object.asyncIter (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/lib.js:334:3)
        at Environment.getTemplate (/Users/loganblangenois/Documents/Sites/Eleventy/projets/mediakod/node_modules/nunjucks/src/environment.js:321:9)
```

If I remove `Nunjucks.Environment`, build works fine.

**To Reproduce**
1. Create a basic `Eleventy` project
2. Install `Nunjucks` environnement
```js
 const Nunjucks = require('nunjucks');

 // inside your module.exports
 let nunjucksEnvironment = new Nunjucks.Environment(
      new Nunjucks.FileSystemLoader("_includes")
 );

 eleventyConfig.setLibrary("njk", nunjucksEnvironment);
```
3. Create a macro inside `_includes` like :
```nunjucks
{% macro foo(bar) %}
   <h1>{{ bar }}</h1>
{% endmacro %}
```

4. Display your macro inside your `index.njk`
```nunjucks
{% from "macros/foo.njk" import foo %}

{{ foo("bar") }}
```

5. Run `yarn dev`

**Environment:**

- OS and Version: MacOS 11.4
- Eleventy Version : ^0.12.1
- Node version : v15.1.0

**Comment by pdehaan** [comment]
Worked for me in a simple test case from your steps to reproduce; https://github.com/pdehaan/11ty-1935. Since everything is in the root folder of the project, I didn't even need to override the default Nunjucks Environment.

But re-reading the error log, I might have an idea of what the issue might be (if you override the env, i think you need the FULL path to the _includes folder, not the relative path from the `dir.input` folder). Let me create a branch and try testing that theory out right quick.

**Comment by pdehaan** [comment]
Yeah, I think that's it. I created a couple new branches:
- https://github.com/pdehaan/11ty-1935/tree/subfolder Moved the files from the root into a "src" folder to match your error message output.
- https://github.com/pdehaan/11ty-1935/tree/custom-njk Uses a custom Nunjucks environment. I saw your error message, but looks like you can fix it with `new Nunjucks.FileSystemLoader("src/_includes")` (including your input directory name of `src/`).

**Comment by pdehaan** [comment]
TL;DR: I think you need a fully qualified path in your `Nunjucks.FileSystemLoader()` call if your project is in a subfolder:

```diff
 const Nunjucks = require('nunjucks');

 module.exports = function (eleventyConfig) {
   const nunjucksEnvironment = new Nunjucks.Environment(
-    new Nunjucks.FileSystemLoader("_includes")
+    new Nunjucks.FileSystemLoader("src/_includes")
   );

   eleventyConfig.setLibrary("njk", nunjucksEnvironment);

   return {
     dir: {
       input: "src",
       output: "www"
     }
   };
 };
```

**Comment by MKlblangenois** [comment]
Thanks @pdehaan ! Your answer solved my issue ! The problem came from there, I just also had to change the way to call my includes like this :
```diff
- {% include "src/_includes/components/organisms/header.njk" %}
+ {% include "components/organisms/header.njk" %}
```

Thanks !
